lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -116,25 +129,30 @@ newNetworkLayer
     :: forall n. ()
     => BaseUrl
     -> IO (NetworkLayer (Jormungandr n) IO)
-newNetworkLayer url = do
+newNetworkLayer = fmap snd . newNetworkLayer'
+
+-- | Creates a new 'NetworkLayer' connecting to an underlying 'Jormungandr'
+-- backend target. Also returns the internal 'JormungandrLayer' component.
+newNetworkLayer'
+    :: forall n. ()
+    => BaseUrl
+    -> IO (JormungandrLayer n IO, NetworkLayer (Jormungandr n) IO)
+newNetworkLayer' url = do
     mgr <- newManager defaultManagerSettings
-    return $ mkNetworkLayer $ mkJormungandrLayer @n mgr url
+    st <- newMVar mempty
+    let jor = mkJormungandrLayer @n mgr url
+    return (jor, mkNetworkLayer st jor)
 
 -- | Wrap a Jormungandr client into a 'NetworkLayer' common interface.
 mkNetworkLayer
-    :: Monad m
-    => JormungandrLayer n m
+    :: MonadBaseControl IO m
```

Comment:
What's the added benefit of using `MonadBaseControl` here instead of `MonadIO` ?

Reply:
Yeah the lifted-base `modifyMVar` wants this because it's a function that has a callback arg.
I am not totally committed to doing it this way.
My main concern is how to make it easy to thoroughly test the `updateUnstableBlocks` logic.